### PR TITLE
Oracle-Database for Azure CLI support

### PR DESCRIPTION
**Pull request by v-eelhomsi**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [*] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [*] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [*] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [*] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by yonzhan** [comment]
Please fix CI issues

File: src/oracle-database/azext_oracle_database/tests/latest/test_oracle_database.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    try_manual,
+    raise_if,
+    calc_coverage
+)
+
+@try_manual
+def setup_scenario(test, rg):
+    pass
+
+@try_manual
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/tests/latest/test_oracle_database.py:23]: Those are not required
```suggestion
```

File: src/oracle-database/azext_oracle_database/tests/latest/test_oracle_database_adbs.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    try_manual,
+    raise_if,
+    calc_coverage
+)
+
+@try_manual
+def setup_scenario(test, rg):
+    pass
+
+@try_manual
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/tests/latest/test_oracle_database_adbs.py:21]:
```suggestion


```

File: src/oracle-database/azext_oracle_database/tests/latest/test_oracle_database_deletes.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    try_manual,
+    raise_if,
+    calc_coverage
+)
+
+@try_manual
+def setup_scenario(test, rg):
+    pass
+
+@try_manual
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/tests/latest/test_oracle_database_deletes.py:21]:
```suggestion


```

File: src/oracle-database/azext_oracle_database/aaz/latest/oracle_database/autonomous_database/_create.py [hunk 4]
```diff
@@ -0,0 +1,1020 @@
...
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "oracle-database autonomous-database create",
+)
+class Create(AAZCommand):
+    """Create a AutonomousDatabase
+    """
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/aaz/latest/oracle_database/autonomous_database/_create.py:19]: Please add examples for the command usage before release, you can following this [doc](https://azure.github.io/aaz-dev-tools/pages/usage/workspace-editor/#command-examples) to add examples in aaz-dev-tools

File: src/oracle-database/azext_oracle_database/aaz/latest/oracle_database/autonomous_database/autonomous_database_backup/__cmd_group.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command_group(
+    "oracle-database autonomous-database autonomous-database-backup",
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/aaz/latest/oracle_database/autonomous_database/autonomous_database_backup/__cmd_group.py:15]: You can simplify it to
```suggestion
    "oracle-database autonomous-database backup",
```

File: src/oracle-database/azext_oracle_database/aaz/latest/oracle_database/autonomous_database_character_set/__cmd_group.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command_group(
+    "oracle-database autonomous-database-character-set",
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/aaz/latest/oracle_database/autonomous_database_character_set/__cmd_group.py:15]:
```suggestion
    "oracle-database autonomous-database character-set",
```

File: src/oracle-database/azext_oracle_database/aaz/latest/oracle_database/autonomous_database_national_character_set/__cmd_group.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command_group(
+    "oracle-database autonomous-database-national-character-set",
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/aaz/latest/oracle_database/autonomous_database_national_character_set/__cmd_group.py:15]:
```suggestion
    "oracle-database autonomous-database national-character-set",
```

File: src/oracle-database/azext_oracle_database/aaz/latest/oracle_database/oracle_subscription/default/__cmd_group.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command_group(
+    "oracle-database oracle-subscription default",
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/aaz/latest/oracle_database/oracle_subscription/default/__cmd_group.py:15]: Recommend to flatten the `default` sub command group, as there's no additional sub command groups in the `oracle-subscription`
```suggestion
    "oracle-database oracle-subscription",
```

File: src/oracle-database/azext_oracle_database/aaz/latest/oracle_database/cloud_exadata_infrastructure/db_server/__cmd_group.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command_group(
+    "oracle-database cloud-exadata-infrastructure db-server",
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/aaz/latest/oracle_database/cloud_exadata_infrastructure/db_server/__cmd_group.py:15]: I found you are using two similar meaning keywords `database` and `db` while naming the commands. It's better to choice only one.

File: src/oracle-database/azext_oracle_database/tests/latest/test_oracle_database_adbs.py [hunk 10]
```diff
@@ -0,0 +1,36 @@
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/tests/latest/test_oracle_database_adbs.py:4]:
```suggestion
# --------------------------------------------------------------------------------------------
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.
#
# Code generated by aaz-dev-tools
# --------------------------------------------------------------------------------------------
```

File: src/oracle-database/azext_oracle_database/tests/latest/test_oracle_database_adbsoper.py [hunk 11]
```diff
@@ -0,0 +1,18 @@
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/tests/latest/test_oracle_database_adbsoper.py:4]:
```suggestion
# --------------------------------------------------------------------------------------------
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.
#
# Code generated by aaz-dev-tools
# --------------------------------------------------------------------------------------------
```

File: src/oracle-database/azext_oracle_database/tests/latest/test_oracle_database_deletes.py [hunk 12]
```diff
@@ -0,0 +1,37 @@
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/tests/latest/test_oracle_database_deletes.py:4]:
```suggestion
# --------------------------------------------------------------------------------------------
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.
#
# Code generated by aaz-dev-tools
# --------------------------------------------------------------------------------------------
```

File: src/oracle-database/azext_oracle_database/azext_metadata.json [hunk 13]
```diff
@@ -0,0 +1,4 @@
+{
+    "azext.isPreview": true,
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/azext_metadata.json:2]: If you want to release GA version, please remove this line
```suggestion
```

File: src/oracle-database/azext_oracle_database/tests/__init__.py [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+# pylint: disable=import-outside-toplevel
+
+import inspect
```
Comment (kairu-ms) [inline src/oracle-database/azext_oracle_database/tests/__init__.py:13]: Please clean up the code in this file, this's not required any more.

**Comment by kairu-ms** [comment]
Please register your top level command group in [src/service_name.json](https://github.com/Azure/azure-cli-extensions/blob/main/src/service_name.json) file.

**Comment by kairu-ms** [comment]
Please run `azdev linter oracle-database` in your local environment. And provide the short option name as alias for the failed parameters by following this [doc](https://azure.github.io/aaz-dev-tools/pages/usage/workspace-editor/#modify-argument-options)

**Comment by kairu-ms** [comment]
This PR is planed to release on Monday 22 July

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**